src/preamble.js

```diff
@@ -481,6 +481,58 @@ function preRun() {
   callRuntimeCallbacks(__ATPRERUN__);
 }
 
+#if ABORT_ON_WASM_EXCEPTIONS
+// Instrument all the exported functions to:
+// - abort if an unhandled exception occurs
+// - throw an exception if someone tries to call them after the program has aborted
+// Set settings.ABORT_ON_WASM_EXCEPTIONS for more info.
+__ATINIT__.push(function exceptionInstrument() { 
```

Comment:
Why not `addOnInit()`?  Or at last unshift so that it will be the first one run?

Reply:
Whoops, that was .unshift at some point. I was following the way wasm_ctors gets added which is directly into __ATINIT__ hence no addOnInit(). I'm fine either way.